op-challenger/flags/flags.go

```diff
@@ -0,0 +1,79 @@
+package flags
+
+import (
+	"fmt"
+
+	"github.com/urfave/cli"
+
+	opservice "github.com/ethereum-optimism/optimism/op-service"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	opmetrics "github.com/ethereum-optimism/optimism/op-service/metrics"
+	oppprof "github.com/ethereum-optimism/optimism/op-service/pprof"
+	oprpc "github.com/ethereum-optimism/optimism/op-service/rpc"
+	txmgr "github.com/ethereum-optimism/optimism/op-service/txmgr"
+)
+
+const envVarPrefix = "OP_CHALLENGER"
+
+var (
+	// Required Flags
+	L1EthRpcFlag = cli.StringFlag{
+		Name:   "l1-eth-rpc",
+		Usage:  "HTTP provider URL for L1.",
+		EnvVar: opservice.PrefixEnvVar(envVarPrefix, "L1_ETH_RPC"),
+	}
+	RollupRpcFlag = cli.StringFlag{
+		Name:   "rollup-rpc",
+		Usage:  "HTTP provider URL for the rollup node.",
+		EnvVar: opservice.PrefixEnvVar(envVarPrefix, "ROLLUP_RPC"),
+	}
+	L2OOAddressFlag = cli.StringFlag{
+		Name:   "l2oo-address",
+		Usage:  "Address of the L2OutputOracle contract.",
+		EnvVar: opservice.PrefixEnvVar(envVarPrefix, "L2OO_ADDRESS"),
+	}
+	DGFAddressFlag = cli.StringFlag{
+		Name:   "dgf-address",
+		Usage:  "Address of the DisputeGameFactory contract.",
+		EnvVar: opservice.PrefixEnvVar(envVarPrefix, "DGF_ADDRESS"),
+	}
+	PrivateKeyFlag = cli.StringFlag{
+		Name:   "private-key",
+		Usage:  "The private key to use with the service. Must not be used with mnemonic.",
+		EnvVar: opservice.PrefixEnvVar(envVarPrefix, "PRIVATE_KEY"),
+	}
+)
+
+var requiredFlags = []cli.Flag{
+	L1EthRpcFlag,
+	RollupRpcFlag,
+	L2OOAddressFlag,
+	DGFAddressFlag,
+}
+
+var optionalFlags = []cli.Flag{
+	PrivateKeyFlag,
+}
+
+func init() {
+	requiredFlags = append(requiredFlags, oprpc.CLIFlags(envVarPrefix)...)
```

Comment:
Should be optional. I'm going to change the proposer to follow this convention.

Reply:
Gotcha, thanks!